### DISCUSSION TITLE
Introduction to System Information Extractor

### DISCUSSION THREAD

**Discussion by yunkon-kim** [cloud-barista/cm-beetle#17] [Show and tell]
### Introduction to System Information Extractor

클라우드 마이그레이션을 수행함에 있어, 기존 컴퓨팅 인프라의 세부 형상 정보를 파악하는 것은 매우 중요한 작업입니다. 어느 수준까지 세부 형상을 파악하는지에 따라 난이도가 상당히 높을 수 있습니다.

그래서 탐색 개발로써, 물리/가상머신의 세부 형상 정보를 파악해 보는 정도의 프로그램이 구현되었습니다. (마이그레이션의 다양한 요소 중, 인프라 측면을 간단히 파악)

자세한 사항 내용은 아래 설명(README) 및 코드를 참고하시기 바랍니다.
 
- [System Information Extractor](https://github.com/cloud-barista/cm-beetle/tree/main/scripts/system-info) (special thanks @seokho-son @Leegyeongjin)

**Comment by niconicodex** [comment]
## 소스 컴퓨팅 환경 정보 수집 활용성 검토

[cm-beetle](https://github.com/cloud-barista/cm-beetle/tree/main) 의 System-Info 기능 확인 완료 하였습니다.
소스 컴퓨팅 환경의 물리/가상 서버의 기본적인 하드웨어적인 정보 수집은 포함되어야

1차 년도에서는 프로세스 단위의 모니터링 메트릭까지는 수집하지 않아도 될 것 같습니다.
예를 들어서 사용자가 운영하는 Legacy 실행 형태의 Nginx 프로세스 가 구동 중이고, 해당 프로세스 컴퓨팅 환경의 인프라 정보는 [System Information Extractor](https://github.com/cloud-barista/cm-beetle/tree/main/scripts/system-info)로 수집되는 정보로 활용 하고, 실제 프로세스가 점유하는 CPU, Memory, DISK I/O 등에 대한 메트릭은 차년도부터 활용 하면 될 것 같습니다.

@yunkon-kim 자세한 구현 목표는 WBS에 맵핑하여 전달 드리겠습니다.
